packages/api-explorer/src/JsonForm/index.jsx

```diff
@@ -1,111 +1,103 @@
-  JSONEditor.defaults.editors.not = notCustomEditor()
-  JSONEditor.defaults.editors.anyOf = anyOfEditor(intl,setFormSubmissionListener)
-
-  JSONEditor.defaults.themes.antdTheme = antdTheme
-
-  // eslint-disable-next-line consistent-return
-  JSONEditor.defaults.resolvers.unshift((scheme) => {
-    // If the schema can be of any type
-    if (
-      (scheme.type === "string" &&
-        scheme.media &&
-        scheme.media.binaryEncoding === "base64") ||
-      scheme.format === "binary"
-    ) {
-      return "base64";
-    }
-  });
-
-  // eslint-disable-next-line consistent-return
-  JSONEditor.defaults.resolvers.unshift((scheme) => {
-    if (scheme.not && scheme.not.type) {
-      return 'not'
-    }
-  })
-
-  // eslint-disable-next-line consistent-return
-  JSONEditor.defaults.resolvers.unshift((scheme) => {
-    if (scheme.anyOf) {
-      return 'anyOf'
-    }
-  })
-}
-
 class JsonForm extends Component {
   constructor(props) {
     super(props);
     this.editor = null;
     this.ref = null;
+    
+    this.state = {
+      error: null,
+      jsonSchema: null
+    }
+    this.unmounted = false
   }
 
-  createEditor(element) {
-    const {intl, onChange, schema, setFormSubmissionListener} = this.props
-    if (this.editor === null) {
-      const schemaToRender = getSchemaToRender(schema)
+  componentDidMount() {
+    const {schema} = this.props
+    resolveReference(schema)
+      .then(convertedSchema => {
+        if (!this.unmounted) {
+          return this.setState({
+            jsonSchema: convertedSchema
+          })
+        }
+        return null
+      }).catch(err => {
+        if (err && !this.unmounted) {
+          return this.setState({error: err.message})
+        }
+        return null
+      })
+  }
+
+  componentWillUnmount() {
+    this.unmounted = true
+  }
 
-      configureJSONEditor(intl, setFormSubmissionListener)
+  createEditor(element, jsonSchema) {
+    const {intl, onChange, setFormSubmissionListener, title} = this.props
+    if (this.editor === null) {
+      configureJsonEditor(JSONEditor, intl, setFormSubmissionListener)
       this.editor = new JSONEditor(element, {
-        schema: schemaToRender,
-        show_opt_in: true,
+        schema: {
+          ...jsonSchema,
+          title
+        },
+        show_opt_in: false,
         prompt_before_delete: false,
         form_name_root:"",
-        theme: "antdTheme"
+        iconlib: 'fontawesome5',
+        theme: 'antdTheme'
       });
       this.editor.on('change', () => onChange(this.editor.getValue()))
     }
   }
 
   render() {
     const {onSubmit} = this.props
-    return (
-      <form
-        ref={r => {
-          this.createEditor(r);
-        }}
-        onSubmit={(e) => {
-          e.preventDefault()
-          onSubmit()
-        }}
-      >
-        <button type="submit" style={{ display: "none" }} />
-      </form>
-    );
+    const {error, jsonSchema} = this.state
+    if (error) {
+      return <span>{error}</span>
```

Comment:
May we could add an antd alert to display the error, what do you think?

Reply:
Done